OpenActive.Server.NET/OpenBookingHelper/OrderCalculations/OrderCalculations.cs

```diff
@@ -1,16 +1,39 @@
 ﻿using OpenActive.NET;
 using System;
 using System.Collections.Generic;
-using System.Text;
 using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
 
 namespace OpenActive.Server.NET.OpenBookingHelper
 {
     public static class OrderCalculations
     {
+        private static readonly IDictionary<string, PropertyInfo> PersonAttributes;
+
+        static OrderCalculations()
+        {
+            var attributes = from property in typeof(Person).GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                             where property.DeclaringType == typeof(Person)
+                             let name = property.GetCustomAttributes<DataMemberAttribute>().SingleOrDefault()?.Name
+                             where name != null
+                             select new { name, property };
+            PersonAttributes = attributes.ToDictionary(t => t.name, t => t.property);
+        }
+
         public static void ValidateAttendeeDetails(OrderItem requestOrderItem, OrderItem responseOrderItem)
         {
-            // TODO: Add attendee errors
+            if (responseOrderItem?.AttendeeDetailsRequired == null || responseOrderItem.Attendee == null)
+                return;
+
+            var values = (from uri in responseOrderItem.AttendeeDetailsRequired
+                          let name = uri.ToString().Split('/').Last()
```

Comment:
@nickevansuk have we implemented some specific way of doing the name comparison that doesn't involve `uri.ToString().Split('/').Last()`?

Reply:
that's the logic we have most commonly implemented